modules/tensor_mechanics/src/materials/SingleVariableReturnMappingSolution.C

```diff
@@ -103,11 +110,11 @@ SingleVariableReturnMappingSolution::returnMappingSolve(const Real effective_tri
     if (!internalSolve(effective_trial_stress, scalar, iter_output_ptr))
     {
       if (iter_output_ptr)
-        mooseError(iter_output_ptr->str());
+        throw MooseException(iter_output_ptr->str());
```

Comment:
Not in the currently implemented models. I have seen it with my own. Sometimes it does through a Petsc error along the lines `Object isn't in correct state` which is related to #10662, but usually it works. I figure it's a lot better to have this than killing the solution.

Reply:
Yeah, I'm fine with making the change.  In my experience, throwing an exception doesn't always result in cutting the time step like I'd like it to.  That's really a separate issue from this. The worst it can do is die with an error like it does now.